packages/design-system-react-native/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,171 @@
+    const currentSelected = isControlled
+      ? Boolean(isSelected)
+      : internalSelected;
+
+    // Animation values
+    const scaleAnim = useRef(new Animated.Value(1)).current;
+    const iconAnim = useRef(
+      new Animated.Value(currentSelected ? 1 : 0),
+    ).current;
+
+    // Sync icon opacity whenever selection changes
+    useEffect(() => {
+      Animated.timing(iconAnim, {
+        toValue: currentSelected ? 1 : 0,
+        duration: 300,
+        easing: Easing.ease,
+        useNativeDriver: true,
+      }).start();
+    }, [currentSelected, iconAnim]);
+
+    // Bounce effect when toggling
+    const animateScale = useCallback(() => {
+      Animated.sequence([
+        Animated.timing(scaleAnim, {
+          toValue: 1.15,
+          duration: 100,
+          easing: Easing.ease,
+          useNativeDriver: true,
+        }),
+        Animated.timing(scaleAnim, {
+          toValue: 1,
+          duration: 100,
+          easing: Easing.ease,
+          useNativeDriver: true,
+        }),
+      ]).start();
+    }, [scaleAnim]);
+
+    // Press handler: update state, notify parent, then bounce
+    const handlePress = () => {
+      if (isDisabled) {
+        return;
+      }
+      const next = !currentSelected;
+      if (!isControlled) {
+        setInternalSelected(next);
+      }
+      onChange?.(next);
+      animateScale();
+    };
+
+    useImperativeHandle(ref, () => ({ toggle: handlePress }), [handlePress]);
+
+    const tw = useTailwind();
+    const twContainerClassNames = `flex-row items-center ${
+      isDisabled ? 'opacity-50' : 'opacity-100'
+    } ${twClassName}`;
+
+    const getCheckboxContainerStyle = useCallback(
+      (pressed: boolean): string => {
+        const baseBg = currentSelected
+          ? 'bg-primary-default'
+          : 'bg-background-default';
+        let baseBorder = 'border-border-default';
+        if (currentSelected) {
+          baseBorder = 'border-primary-default';
+        } else if (isInvalid) {
+          baseBorder = 'border-error-default';
+        }
+        const pressedBg = currentSelected
+          ? 'bg-primary-defaultPressed'
+          : 'bg-background-defaultPressed';
+        let pressedBorder = 'border-border-default';
+        if (currentSelected) {
+          pressedBorder = 'border-primary-defaultPressed';
+        } else if (isInvalid) {
+          pressedBorder = 'border-error-default';
+        }
+        return pressed
+          ? `${pressedBg} ${pressedBorder}`
+          : `${baseBg} ${baseBorder}`;
+      },
+      [currentSelected, isInvalid],
+    );
+
+    return (
+      <Pressable
+        onPress={handlePress}
+        accessible
+        accessibilityRole="checkbox"
+        accessibilityState={{
+          checked: currentSelected,
+          disabled: isDisabled,
+        }}
+        accessibilityLabel={typeof label === 'string' ? label : undefined}
+        style={[tw`${twContainerClassNames}`, style as StyleProp<ViewStyle>]}
```

Comment:
By moving the twrnc hook `tw`(useTailwind) closer to where the Tailwind classnames are applied, we can take advantage of Tailwind IntelliSense. This could improve the developer experience by providing class suggestions and autocomplete, and may also help simplify the code by making the usage of tw more direct and readable.

```suggestion
        style={[twContainerClassNames, style as StyleProp<ViewStyle>]}
```